lib/facility_generator.rb

```diff
@@ -0,0 +1,29 @@
+class FacilityGenerator
+    def create_facilities(facilities_data)
+        facilities_data.map do |data|
+            if data[:state] == 'CO'
+                data[:name] = data[:dmv_office]
+                data.delete(:dmv_office)
+                data[:address] = "#{data[:address_li]}, #{data[:address_1]}, #{data[:city]}, #{data[:state]} #{data[:zip]}"
+                data[:services] = []
+                if data[:services_p].include?('registration')
+                    data[:services] << ['Vehicle Registration']
+                end
+                Facility.new(data)
+            elsif data[:state] == 'NY'
+                data[:name] = data[:office_name]
+                data.delete(:office_name)
+                data[:address] = "#{data[:street_address_line_1]}, #{data[:city]}, #{data[:state]} #{data[:zip_code]}"
+                data[:hours] = "Mon-Fri #{data[monday_beginning_hours]}-#{data[monday_ending_hours]}"
+                Facility.new(data)
+            else data[:state] == 'MO'
+                data[:address] = "#{data[:address1]}, #{data[:city]}, #{data[:state]} #{data[:zipcode]}"
+                data[:hours] = data[:daysopen]
+                Facility.new(data)
+            end
+            # is key mapping more efficient than this?
+            # pros : looks cleaner, easier to read
+            # cons : still have to write out new api keys
+        end
```

Comment:
Making a new hash with these keys is not necessarily more efficient than this, but might be a bit more readable. One major way to break this up and make it more readable would be to introduce helper methods based on the state:
```
if data[:state] == "MO"
     generate_mo_facility(data)
elseif data[:state] == "NY"
     generate_ny_facility(data)
...
```
and then your helper method could look like this:
```
def generate_mo_facility(data)
    facility_data = {
        name: data[:dmv_office]
        address: "#{data[:address_li]}, #{data[:address_1]}, #{data[:city]}, #{data[:state]} #{data[:zip]}"
        ....
    }
    Facility.new(facility_data)
}
```